agent/src/main/java/com/teamscale/jacoco/agent/upload/delay/DelayedUploader.java

```diff
@@ -83,20 +82,24 @@ public String describe() {
 	 * Sets the commit to upload the XMLs to and asynchronously triggers the upload of all cached XMLs. This method
 	 * should only be called once.
 	 */
-	public synchronized void setCommitAndTriggerAsynchronousUpload(String commit) {
+	public synchronized void setCommitAndTriggerAsynchronousUpload(T information) {
 		if (wrappedUploader == null) {
-			wrappedUploader = wrappedUploaderFactory.apply(commit);
-			logger.info("Commit to upload to has been found: {}. Uploading any cached XMLs now to {}", commit,
+			wrappedUploader = wrappedUploaderFactory.apply(information);
+			logger.info("Commit to upload to has been found: {}. Uploading any cached XMLs now to {}", information,
 					wrappedUploader.describe());
 			executor.execute(this::uploadCachedXmls);
 		} else {
 			logger.error("Tried to set upload commit multiple times (old uploader: {}, new commit: {})." +
-					" This is a programming error. Please report a bug.", wrappedUploader.describe(), commit);
+					" This is a programming error. Please report a bug.", wrappedUploader.describe(), information);
 		}
 	}
 
 	private void uploadCachedXmls() {
 		try {
+			if (!Files.isDirectory(cacheDir)) {
```

Comment:
For me the cacheDir was never set when the callback fired, so maybe looks like another Java14 issue